### PR TITLE
iox-#496 improvre coverage for class port_pool in roudi folder

### LINKED ISSUE #496: Add unit test for iceoryx-posh
## For iceoryx is more testing needed to achieve good code coverage

add more unit/module tests for following files

- [x] capro/service_description.cpp 
- [ ] mepoo/shared_chunk.cpp 
- [ ] mepoo/ mem_pool.cpp 
- [ ] mepoo/memory_manager.cpp 
- [ ] mepoo/ mepoo_config.cpp 
- [ ] runtime/shared_memory_user.cpp 
- [ ] runtime/posh_runtime_single_process.cpp 
- [ ] runtime/posh_runtime.cpp 
- [x] runtime/node.cpp 
- [ ] gateway/gateway_base.cpp 
- [ ] gateway/gateway_config.cpp 
- [ ] gateway/toml_gateway_config_parser.cpp 
- [ ] roudi/port_pool.cpp 
- [ ] roudi/roudi_process.cpp 
- [ ] roudi/ roudi_config.cpp 
- [ ] roudi/port_manager.cpp 
- [ ] roudi/roudi_cmd_line_parser.cpp 
- [ ] roudi/roudi_cmd_line_parser_config_file_option.cpp 
- [ ] roudi/memory/memory_block.cpp 
- [ ] roudi/memory/roudi_memory_manager.cpp 
- [ ] roudi/memory/memory_provider.cpp 
- [ ] roudi/application/ iceoryx_roudi_app.cpp 

- [x] Define a default constructor ServiceDescription::ClassHash 

reference for code-coverage
https://codecov.io/gh/eclipse-iceoryx/iceoryx?branch=master

### PR DESCRIPTION
**Pull request by nihalchari**
Signed-off-by: Chari Nehal Dattaram (RBEI/EBB1) <NehalDattaram.Chari@in.bosch.com>

## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Notes for Reviewer

purpose of this PR is to increase coverage for class port_pool.cpp

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- Partly closes #496

### CODE REVIEW

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using ::testing::Return;
+
+namespace example
+{
+namespace test
+{
+class PortPool_test : public Test
+{
+  public:
+    iox::roudi::PortPoolData* portPoolData = new iox::roudi::PortPoolData{};
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:31]: You need to delete this, otherwise there is a memleak

Comment (dkroenke) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:31]: Update:
I was wondering why PortPoolData was created on heap, i think this is not necessary here. We should avoid heap in test too if possible.
My advise would be to restructure the test:

1. remove the placement new for PortPoolData and add it directly to the test class as member
2. use the PortPoolData member to add a new PortPool member with the name `sut` to make clear that this is your "system under test". Use the previously created PortPoolData member to initialize it.
3. use `sut` in your test case functions to fill the vector and test it.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    iox::Error errorHandlerType;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0; i < iox::MAX_NODE_NUMBER; ++i)
+    {
+        portPoolData->m_nodeMembers.insert("processName", "nodeName", i);
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:46]: please use `addNodeData` function like below

**Review by dkroenke — CHANGES_REQUESTED** [review]
Please rework the test to remove the placement new for PortPoolData.
Question: Is this the only test which will come in this PR or are there additional tests coming?   
I'm asking because the other member functions are not tested yet.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
```
Comment (elfenpiff) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:1]: What is the goal of this pull request? You are testing only a single aspect of one method. 

For me it seems that this pull request is incomplete since all the other methods are not tested at all. If there are some reasons for adding just one single test maybe they can be stated in the "Notes for the Reviewer" section in the pr template.

**Review by elfenpiff — DISMISSED** [review]
Clarify intention of the pull request? Why is there only one aspect of one method tested?

Is this pull request incomplete or are the other unit tests still coming?

**Comment by dkroenke** [comment]
@nihalchari please link the issue

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 2 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:46]: m_nodeMembers is iox::roudi::FixedPositionContainer which doest not have addNodeData function

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0; i < iox::MAX_NODE_NUMBER; ++i)
+    {
+        portPoolData->m_nodeMembers.insert("processName", "nodeName", i);
+    }
+
+    iox::roudi::PortPool portPool(*portPoolData);
+    portPool.addNodeData("processName", "nameName", 999u);
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:50]: > Please rework the test to remove the placement new for PortPoolData.
> Question: Is this the only test which will come in this PR or are there additional tests coming?
> I'm asking because the other member functions are not tested yet.

Only one test was added considering coverage report, but yeah if other functionality is not tested then i can add that too
@elfenpiff FYI

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 2 above)
Comment (dkroenke) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:46]: Yes i see that, but why are you inserting the nodename directly into the container instead of using `addNodeData` from the Portpool?
See comment: https://github.com/eclipse-iceoryx/iceoryx/pull/516#discussion_r562094828

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: added

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+{
+    auto errorHandlerCalled{false};
+    iox::Error errorHandlerType;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0; i < iox::MAX_NODE_NUMBER; ++i)
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+#include "iceoryx_posh/internal/roudi/port_pool_data.hpp"
+#include "iceoryx_posh/internal/runtime/node_data.hpp"
+#include "iceoryx_posh/roudi/port_pool.hpp"
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace ::testing;
+using ::testing::Return;
+
+namespace example
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: namespace example removed

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/internal/roudi/port_pool_data.hpp"
+#include "iceoryx_posh/internal/runtime/node_data.hpp"
+#include "iceoryx_posh/roudi/port_pool.hpp"
+#include <gmock/gmock.h>
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: replaced

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 1 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0U; i < iox::MAX_NODE_NUMBER; ++i)
+    {
+        portPoolData.m_nodeMembers.insert("processName", "nodeName", i);
+    }
+
+    iox::roudi::PortPool sut(portPoolData);
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:46]: This can be done in the test class

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    iox::Error errorHandlerType;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0U; i < iox::MAX_NODE_NUMBER; ++i)
+    {
+        portPoolData.m_nodeMembers.insert("processName", "nodeName", i);
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:43]: please use `sut.addNodeData` here

**Review by dkroenke — CHANGES_REQUESTED** [review]

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 10 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 9 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 2 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

**Comment by nihalchari** [comment]
I will add more test for remaining functionalities

**Review by elfenpiff — COMMENTED** [review]
@nihalchari If you add the license statement I will approve this pull request.
I think the tests are incomplete but the provided code is fine.

File: iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp [hunk 11]
```diff
@@ -47,7 +47,7 @@ class PoshRuntimeSingleProcess_test : public Test
 
 TEST_F(PoshRuntimeSingleProcess_test, ConstructorPoshRuntimeSingleProcessIsSuccess)
 {
-    iox::RouDiConfig_t defaultRouDiConfig = iox::RouDiConfig_t().setDefaults();
+    iox::RouDiConfig_t defaultRouDiConfig = iox::RouDiConfig_t().optimize();
```
Comment (nihalchari) [inline iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp:50]: this was changed as to improve code coverage for https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/source/roudi/roudi_config.cpp

Comment (dkroenke) [reply iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp:50]: > this was changed as to improve code coverage for https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/source/roudi/roudi_config.cpp

It is dangerous to increase the unittest coverage for a class which is not tested here because this can lead to wrong assumptions. RouDiConfig is not the test subject here and should be tested in a separate testsuite.
The `optimize()` method is not intended to be used for creating a default roudi config.
https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp:50]: okay i will revert this changes, could you please take a look in rest of the test?

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 12]
```diff
@@ -0,0 +1,604 @@
...
+
+using namespace ::testing;
+using ::testing::Return;
+using namespace iox::capro;
+
+namespace iox
+{
+namespace test
+{
+static constexpr uint32_t DEFAULT_DEVICE_ID{0u};
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:31]: Please initialize here with values other than null to ensure that the passed parameters are correctly taken.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 13]
```diff
@@ -0,0 +1,604 @@
...
+{
+    auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+
+    sut.removeConditionVariableData(conditionVariableData.value());
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetServiceRegistryChangeCounterIsSuccessful)
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:597]: This tests against the initial value of the `serviceRegistryChangeCounter` so the name needs to be adapted, would be good to have a test that return something else than null.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 14]
```diff
@@ -0,0 +1,604 @@
...
+TEST_F(PortPool_test, GetConditionVariableDataListWhenEmptyIsSuccessful)
+{
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetConditionVariableDataListCompletelyFilledIsSuccessful)
+{
+    for (uint32_t i = 0; i < MAX_NUMBER_OF_CONDITION_VARIABLES; ++i)
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:577]: i is unsigned

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 15]
```diff
@@ -0,0 +1,604 @@
...
+    Error errorHandlerType;
+    auto errorHandlerGuard =
+        ErrorHandler::SetTemporaryErrorHandler([&](const Error error, const std::function<void()>, const ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0U; i <= MAX_PROCESS_NUMBER; ++i)
+    {
+        auto pplicationData = sut.addApplicationPort(m_applicationName);
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:474]: typo

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 16]
```diff
@@ -0,0 +1,604 @@
...
+TEST_F(PortPool_test, GetInterfacePortDataListWhenEmptyIsSuccessful)
+{
+    auto interfacePortDataList = sut.getInterfacePortDataList();
+
+    ASSERT_EQ(interfacePortDataList.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetInterfacePortDataListCompletelyFilledIsSuccessful)
+{
+    for (uint32_t i = 0; i < MAX_INTERFACE_NUMBER; ++i)
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:419]: i is unsigned

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 17]
```diff
@@ -0,0 +1,604 @@
...
+TEST_F(PortPool_test, GetSubscriberPortDataListWhenEmptyIsSuccessful)
+{
+    auto nodeDataList = sut.getSubscriberPortDataList();
+
+    ASSERT_EQ(nodeDataList.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetSubscriberPortDataListCompletelyFilledIsSuccessful)
+{
+    for (uint32_t i = 0; i < MAX_SUBSCRIBERS; ++i)
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:333]: i is unsigned

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 18]
```diff
@@ -0,0 +1,604 @@
...
+{
+    auto errorHandlerCalled{false};
+    Error errorHandlerType;
+    auto errorHandlerGuard =
+        ErrorHandler::SetTemporaryErrorHandler([&](const Error error, const std::function<void()>, const ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0; i <= MAX_SUBSCRIBERS; ++i)
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:299]: i is unsigned

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 19]
```diff
@@ -0,0 +1,604 @@
...
+{
+    auto errorHandlerCalled{false};
+    Error errorHandlerType;
+    auto errorHandlerGuard =
+        ErrorHandler::SetTemporaryErrorHandler([&](const Error error, const std::function<void()>, const ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0; i <= MAX_PUBLISHERS; ++i)
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:185]: i is unsigned

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 20]
```diff
@@ -0,0 +1,604 @@
...
+}
+
+TEST_F(PortPool_test, GetServiceRegistryChangeCounterIsSuccessful)
+{
+    auto serviceCounter = sut.serviceRegistryChangeCounter();
+
+    ASSERT_EQ(serviceCounter->load(), 0U);
+}
+} // namespace test
+} // namespace iox
```
Comment (dkroenke) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:604]: missing endline

**Review by dkroenke — CHANGES_REQUESTED** [review]

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 20 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: added

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 13 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done, changed name and added new test

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp (hunk 12 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 21]
```diff
@@ -0,0 +1,604 @@
...
+    roudi::PortPoolData m_portPoolData;
+    roudi::PortPool sut{m_portPoolData};
+
+    ServiceDescription m_serviceDescription{"service1", "instance1"};
+    ProcessName_t m_applicationName{"AppName"};
+    ProcessName_t m_processName{"processName"};
+    ProcessName_t m_nodeName{"nodeName"};
+    const uint64_t m_nodeDeviceId = 999U;
+    mepoo::MemoryManager m_memoryManager;
+    popo::PublisherOptions m_publisherOptions;
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: done

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 22]
```diff
@@ -0,0 +1,604 @@
...
+    ASSERT_TRUE(errorHandlerCalled);
+    ASSERT_EQ(errorHandlerType, Error::kPORT_POOL__NODELIST_OVERFLOW);
+}
+
+TEST_F(PortPool_test, GetNodeDataListIsSuccessful)
+{
+    sut.addNodeData(m_processName, m_nodeName, m_nodeDeviceId);
+    auto nodeDataList = sut.getNodeDataList();
+
+    ASSERT_EQ(nodeDataList.size(), 1U);
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:101]: done

File: iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp (hunk 11 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp]: reverted changes

**Review by dkroenke — APPROVED** [review]

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 23]
```diff
@@ -0,0 +1,625 @@
...
+    for (uint32_t i = 1U; i <= MAX_NUMBER_OF_CONDITION_VARIABLES; ++i)
+    {
+        auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+        if (!conditionVariableData.has_error())
+        {
+            conditionVariableContainer.push_back(conditionVariableData.value());
+        }
+    }
+
+    ASSERT_EQ(conditionVariableContainer.size(), MAX_NUMBER_OF_CONDITION_VARIABLES);
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:550]: Please use ``EXPECT_EQ``.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 24]
```diff
@@ -0,0 +1,625 @@
...
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0U; i <= MAX_NUMBER_OF_CONDITION_VARIABLES; ++i)
+    {
+        auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+    }
+
+    ASSERT_TRUE(errorHandlerCalled);
+    ASSERT_EQ(errorHandlerType, Error::kPORT_POOL__CONDITION_VARIABLE_LIST_OVERFLOW);
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:569]: Please use ``EXPECT_*``.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 25]
```diff
@@ -0,0 +1,625 @@
...
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetServiceRegistryChangeCounterReturnsZeroAsInitialValue)
+{
+    auto serviceCounter = sut.serviceRegistryChangeCounter();
+
+    ASSERT_EQ(serviceCounter->load(), 0U);
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:613]: Please use ``EXPECT_EQ``.

**Review by FerdinandSpitzschnueffler — CHANGES_REQUESTED** [review]

**Comment by elfenpiff** [comment]
@nihalchari Please refrain from commit messages like: `fix review points`. Based on the git message we should see what was done in the code and fix review points is not helpful as a message since we do not have access at all to our comments you worked on in this particular commit.
A better way is just to state what you have actually done. Maybe something like: `added missing test for case X` or `added U postfix for unsigned number` or `added missing tags in documentation` ... etc...

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 26]
```diff
@@ -0,0 +1,625 @@
...
+    for (uint32_t i = 0U; i <= MAX_NODE_NUMBER; ++i)
+    {
+        sut.addNodeData(m_processName, m_nodeName, i);
+    }
+
+    ASSERT_TRUE(errorHandlerCalled);
+    ASSERT_EQ(errorHandlerType, Error::kPORT_POOL__NODELIST_OVERFLOW);
+}
+
+TEST_F(PortPool_test, GetNodeDataListIsSuccessful)
```
[the remark this answers is not in the document]
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:93]: Separate test was added so that we can have one test for one functionality.

**Comment by nihalchari** [comment]
> 
> 
> @nihalchari Please refrain from commit messages like: `fix review points`. Based on the git message we should see what was done in the code and fix review points is not helpful as a message since we do not have access at all to our comments you worked on in this particular commit.
> A better way is just to state what you have actually done. Maybe something like: `added missing test for case X` or `added U postfix for unsigned number` or `added missing tags in documentation` ... etc...

changed commit messages

**Review by dkroenke — DISMISSED** [review]

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 27]
```diff
@@ -0,0 +1,625 @@
...
+using namespace ::testing;
+using ::testing::Return;
+using namespace iox::capro;
+
+namespace iox
+{
+namespace test
+{
+static constexpr uint32_t DEFAULT_DEVICE_ID{20u};
+static constexpr uint32_t DEFAULT_MEMORY_TYPE{100u};
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: Looks like it's done.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 28]
```diff
@@ -0,0 +1,625 @@
...
+    roudi::PortPoolData m_portPoolData;
+    roudi::PortPool sut{m_portPoolData};
+
+    ServiceDescription m_serviceDescription{"service1", "instance1"};
+    ProcessName_t m_applicationName{"AppName"};
+    ProcessName_t m_processName{"processName"};
+    ProcessName_t m_nodeName{"nodeName"};
+    const uint64_t m_nodeDeviceId = 999U;
+    mepoo::MemoryManager m_memoryManager;
+    popo::PublisherOptions m_publisherOptions{10, "nodeName"};
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: Looks like it's done.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 29]
```diff
@@ -0,0 +1,625 @@
...
+class PortPool_test : public Test
+{
+  public:
+    roudi::PortPoolData m_portPoolData;
+    roudi::PortPool sut{m_portPoolData};
+
+    ServiceDescription m_serviceDescription{"service1", "instance1"};
+    ProcessName_t m_applicationName{"AppName"};
+    ProcessName_t m_processName{"processName"};
+    ProcessName_t m_nodeName{"nodeName"};
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: Looks like it's done.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 30]
```diff
@@ -0,0 +1,625 @@
...
+    const uint64_t m_nodeDeviceId = 999U;
+    mepoo::MemoryManager m_memoryManager;
+    popo::PublisherOptions m_publisherOptions{10, "nodeName"};
+    popo::SubscriberOptions m_subscriberOptions{
+        iox::popo::SubscriberPortData::ChunkQueueData_t::MAX_CAPACITY, 10, "nodeName"};
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp]: Looks like it's done.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 31]
```diff
@@ -0,0 +1,625 @@
...
+    iox::mepoo::MemoryInfo m_memoryInfo{DEFAULT_DEVICE_ID, DEFAULT_MEMORY_TYPE};
+};
+
+TEST_F(PortPool_test, AddNodeDataIsSuccessful)
+{
+    auto nodeData = sut.addNodeData(m_processName, m_nodeName, m_nodeDeviceId);
+
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:54]: Looks like it's done.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 32]
```diff
@@ -0,0 +1,625 @@
...
+    sut.removeSubscriberPort(subscriberPort.value());
+    auto subscriberPortDataList = sut.getSubscriberPortDataList();
+
+    ASSERT_EQ(subscriberPortDataList.size(), 0U);
+}
+
+TEST_F(PortPool_test, AddInterfacePortIsSuccessful)
+{
+    auto interfacePortData = sut.addInterfacePort(m_applicationName, Interfaces::INTERNAL);
+
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:355]: Looks like it's done.

File: iceoryx_posh/test/moduletests/test_roudi_portpool.cpp [hunk 33]
```diff
@@ -0,0 +1,625 @@
...
+    sut.removeInterfacePort(interfacePort.value());
+    auto interfacePortDataList = sut.getInterfacePortDataList();
+
+    ASSERT_EQ(interfacePortDataList.size(), 0U);
+}
+
+TEST_F(PortPool_test, AddApplicationPortIsSuccessful)
+{
+    auto applicationPortData = sut.addApplicationPort(m_applicationName);
+
```
[the remark this answers is not in the document]
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_roudi_portpool.cpp:431]: Looks like it's done.

**Review by marthtz — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

File: iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp (hunk 11 above)
Comment (nihalchari) [reply iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp]: for roudi_config.hpp i am try to add below test, 
TEST(RoudiConfig_test, OptimizeIsSucessfull)
{
    iox::config::RouDiConfig sut;
    auto optimizedRoudiConfig = sut.optimize();
    EXPECT_EQ(sut, optimizedRoudiConfig);
}

this requires to add operator= in struct RouDiConfig(). 

@dkroenke  Is a good way to continue?
